toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -306,20 +306,39 @@ func (c *Chroot) AddFiles(filesToCopy ...FileToCopy) (err error) {
 	return addFilesToDestination(c.rootDir, filesToCopy...)
 }
 
-func addFilesToDestination(destDir string, filesToCopy ...FileToCopy) error {
+func addFileToDestination(destDir string, fileToCopy FileToCopy) error {
+	dest := filepath.Join(destDir, fileToCopy.Dest)
+	logger.Log.Debugf("Copying '%s' to '%s'", fileToCopy.Src, dest)
+
+	var err error
+	if fileToCopy.Permissions != nil {
+		err = file.CopyAndChangeMode(fileToCopy.Src, dest, os.ModePerm, *fileToCopy.Permissions)
+	} else {
+		err = file.Copy(fileToCopy.Src, dest)
+	}
+
+	if err != nil {
+		logger.Log.Errorf("Error copying file '%s'", fileToCopy.Src)
+		return err
+	}
+
+	return nil
+}
+
+func AddFilesToDestination(destDir string, filesToCopy []FileToCopy) error {
```

Comment:
You could implement this function like:

```golang
func AddFilesToDestination(destDir string, filesToCopy []FileToCopy) error {
    return addFilesToDestination(destDir, filesToCopy...)
}
```

Of course, callers could do this as well.

Reply:
good point - left as-is and updated the caller.